middle_end/flambda/simplify/basic/continuation_in_env.mli

```diff
@@ -18,21 +18,23 @@
 
 type t =
   | Linearly_used_and_inlinable of {
-      arity : Flambda_arity.With_subkinds.t;
+      params : Kinded_parameter.t list;
       (** To avoid re-opening name abstractions, we store the opened
           parameters and handler here.  This is only correct because the
           inlining we perform is linear. *)
-      params : Kinded_parameter.t list;
       handler : Flambda.Expr.t;
       (** [free_names_of_handler] includes entries for any occurrences of the
           [params] in the [handler]. *)
       free_names_of_handler : Name_occurrences.t;
       (** [cost_metrics_of_handler] is the size of the handler. *)
       cost_metrics_of_handler : Flambda.Cost_metrics.t;
     }
-  | Other of {
+  | Non_inlinable of {
+      params : Kinded_parameter.t list;
+      handler : Flambda.Expr.t Or_unknown.t;
+    }
```

Comment:
Ah, these could deserve a doc comment similar to the one above, or at least explaining why it's safe/reasonable to sotre the parameters and body without the binder.

Reply:
Ah, now this is a good point: in fact the above representation was kind of bad.  I've updated it in the latest commit.  I've also updated the existing comment about name abstractions which wasn't correct.